packages/transformers/react-refresh-wrap/src/helpers/helpers.js

```diff
@@ -37,7 +38,16 @@ var enqueueUpdate = debounce(function () {
 // MIT License - Copyright (c) Facebook, Inc. and its affiliates.
 
 module.exports.prelude = function (module) {
+  window.__REACT_REFRESH_VERSION_TRANSFORMER = version;
   window.$RefreshReg$ = function (type, id) {
+    if (
+      window.__REACT_REFRESH_VERSION_TRANSFORMER !==
+      window.__REACT_REFRESH_VERSION_RUNTIME
+    ) {
+      throw new Error(
+        `react-refresh versions did not match between transformer and runtime. Please check your dependencies. Transformer: ${window.__REACT_REFRESH_VERSION_TRANSFORMER}, Runtime: ${window.__REACT_REFRESH_VERSION_RUNTIME}`,
+      );
+    }
```

Comment:
Maybe better to enforce this at the package manager level rather than at runtime? Eg yarn constraints or resolutions

Reply:
Yeah the challenge here is that multiple repos were onboarding to parcel and didn't understand the problem. Handling this at build time would be more ideal I think, not sure if there's prior art of managing dependencies like this in Parcel though.